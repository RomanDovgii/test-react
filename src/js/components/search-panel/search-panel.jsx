import React, {createRef} from "react";
import {connect} from "react-redux";
import {searchUsers, updateCurrentUsers, updatePagesCount} from "../../store/action/action";

const SearchPanel = ({isLoading, onSearchClick}) => {
  const searchRef = createRef();

  return (
    <form className="controls__filtration">
      <input className="controls__search" type="text" name="filtration input" placeholder="Введите данные поиска" ref={searchRef}/>
      <button className="controls__find" type="submit" onClick={(evt) => {
        evt.preventDefault();

        if (!isLoading) {
          const searchString = searchRef.current.value;
          onSearchClick(searchString);
        }
      }}>Найти</button>
    </form>
  );
};

const mapStateToProps = (state) => ({
  isLoading: state.isLoading
});

const mapDispatchToProps = (dispatch) => ({
  onSearchClick(searchString) {
    dispatch(searchUsers(searchString));
    dispatch(updateCurrentUsers());
    dispatch(updatePagesCount());
  }
});

export {SearchPanel};
export default connect(mapStateToProps, mapDispatchToProps)(SearchPanel);
