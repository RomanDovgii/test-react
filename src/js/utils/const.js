export const MAXIMUM_USERS_PER_PAGE = 50;

export const AppRoute = {
  SELECT: `/select`,
  TABLE: `/table`
};
