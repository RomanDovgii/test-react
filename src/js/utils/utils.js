export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortToTop = (a, b) => {
  switch (true) {
    case a > b:
      return -1;
    case a < b:
      return 1;
    default:
      return 0;
  }
};

export const sortToBot = (a, b) => {
  switch (true) {
    case a < b:
      return -1;
    case a > b:
      return 1;
    default:
      return 0;
  }
};
