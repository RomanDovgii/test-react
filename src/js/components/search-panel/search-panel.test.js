import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {SearchPanel} from "./search-panel";
import {noop} from "../../utils/utils";

it(
    `Should render SearchPanel locked`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <SearchPanel
            isLoading={true}
            onSearchClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);

it(
    `Should render SearchPanel unlocked`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <SearchPanel
            isLoading={false}
            onSearchClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);
