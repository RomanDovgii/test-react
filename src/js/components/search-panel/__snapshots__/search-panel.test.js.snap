// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SearchPanel locked 1`] = `
<form
  className="controls__filtration"
>
  <input
    className="controls__search"
    name="filtration input"
    placeholder="Введите данные поиска"
    type="text"
  />
  <button
    className="controls__find"
    onClick={[Function]}
    type="submit"
  >
    Найти
  </button>
</form>
`;

exports[`Should render SearchPanel unlocked 1`] = `
<form
  className="controls__filtration"
>
  <input
    className="controls__search"
    name="filtration input"
    placeholder="Введите данные поиска"
    type="text"
  />
  <button
    className="controls__find"
    onClick={[Function]}
    type="submit"
  >
    Найти
  </button>
</form>
`;
